### PR TITLE
Add seasonality to CH4 reservoir emissions

### PR DESCRIPTION
**Pull request by msulprizio**
### Name and Institution (Required)

Name: Melissa Sulprizio
Institution: Harvard

### Describe the update

Previously reservoir emissions from the ResME inventory were included every month but there should be none during the cold months. From Delwiche et al. (2022):

>Since individual reservoir modeling for stratification is beyond the scope of ResME, we instead provide degassing estimates for two potential scenarios: (a) maximum possible degassing if reservoirs were stratified year-round and thus T b = (0.5 + Bubdiss) x P in each month, and (b) reservoir is stratified for the 2 months before and after peak reservoir temperature (5 months total), with monthly degassing zero otherwise.

To apply seasonality to the ResME emissions, a monthly mask file has been created to zero emissions above 30N and below 30S outside of the 5 warmest months for that hemisphere.

### Expected changes

This is a zero-difference update with respect to full-chemistry benchmark simulations. 

In carbon and CH4 simulations, CH4 reservoir emissions are reduced outside of the summer months. Changes from 1-month CH4 benchmark simulations for 2019 are included below: 

```
#########################################################################################################
### Emissions totals for species CH4 [Tg]                                                             ###
### Ref = GCC_14.4.0_CH4                                                                              ###
### Dev = GCC_14.5.0_CH4                                                                              ###
#########################################################################################################
                                    Ref                 Dev           Dev - Ref              % diff diffs
CH4 Reservoirs     :          26.549702           14.625304          -11.924399          -44.913492   * 
```

### Reference(s)

- Delwiche, K.B., J.A. Harrison, J.D. Maasakkers, M.P. Sulprizio, J. Worden, D.J. Jacob, and E.M. Sunderland, _Estimating drivers and pathways for hydroelectric reservoir methane emissions using a new mechanistic model,_ J. Geophys. Res. Biogeosciences,127, e2022JG006908, 2022, https://doi.org/10.1029/2022JG006908.

### CODE REVIEW

File: run/GCHP/ExtData.rc.templates/ExtData.rc.carbon [hunk 1]
```diff
@@ -432,6 +432,9 @@ CONUS_MASK_MIRROR 1 N Y - none none MASK ./HcoDir/MASKS/v2024-04/CONUS_Mask_Mirr
 
 # Comment these lines if you are not using any of the CEDS inventories
 CHINA_MASK        1 N Y - none none MASK ./HcoDir/MASKS/v2018-09/China_mask.generic.1x1.nc
+
+# For CH4 reservoir emissions from ResME
+SEASONAL_RES_MASK 1 N Y F%y2-%m2-01T00:00:00 - none none Mask ./HcoDir/CH4/v2024-01/ResME/reservoir_mask.01x01.nc
```
Comment (lizziel) [inline run/GCHP/ExtData.rc.templates/ExtData.rc.carbon:437]: Change 'N' to 'Y' (climatology column) and 'F%y2-%m2' to 'F2022-%m2' (year of climatology in file).

**Review by lizziel — CHANGES_REQUESTED** [review]

File: run/GCHP/ExtData.rc.templates/ExtData.rc.carbon (hunk 1 above)
Comment (msulprizio) [reply run/GCHP/ExtData.rc.templates/ExtData.rc.carbon]: Thanks for catching that. I meant to change the year to 2022 but did not realize about the climatology column. This should now be fixed in https://github.com/geoschem/geos-chem/pull/2473/commits/f25fb1c695369a7813252001a076fd66127ebc6c.

Comment (lizziel) [reply run/GCHP/ExtData.rc.templates/ExtData.rc.carbon]: Yeah, the climatology column is technically not necessary but it skips some logic in the model so is more efficient.

**Review by lizziel — APPROVED** [review]

**Comment by msulprizio** [comment]
After removing an extra dash from ExtData.rc.carbon in https://github.com/geoschem/geos-chem/pull/2473/commits/f2f6e09716e7a44ad4eb78758176e701d44fc2ef, all GCClassic and GCHP integration tests pass

**Review by lizziel — APPROVED** [review]

**Pull request merged**